src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Microsoft.VisualStudio.LanguageServerClient.Razor.csproj

```diff
@@ -19,6 +19,8 @@
   <ItemGroup>
     <PackageReference Include="System.Composition" Version="$(SystemCompositionPackageVersion)" />
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="$(Tooling_MicrosoftCodeAnalysisAnalyzersPackageVersion)" NoWarn="NU1608" />
+    <PackageReference Include="Microsoft.Extensions.Caching.Abstractions" Version="$(MicrosoftExtensionsCachingAbstractions)" />
+    <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="$(MicrosoftExtensionsCachingMemory)" />
```

Comment:
Memory doesn't bring in abstractions?

Reply:
Nope, it complains about missing abstractions (extension fails to boot).

Reply:
Something is wrong here
- https://www.nuget.org/packages/Microsoft.Extensions.Caching.Memory/5.0.0-preview.6.20305.6 shows the latest Caching.Memory version depends on Caching.Abstractions
- same dependency shows up in https://www.nuget.org/packages/Microsoft.Extensions.Caching.Memory/3.1.6
- same dependency shows up in https://www.nuget.org/packages/Microsoft.Extensions.Caching.Memory/3.1.0 too

Was the problem only with src/Razor/src/Microsoft.VisualStudio.RazorExtension/Microsoft.VisualStudio.RazorExtension.csproj❔

Reply:
Just tried it again, and without including `Caching.Abstractions` I get:
```
System.AggregateException: One or more errors occurred. ---> Microsoft.VisualStudio.Composition.CompositionFailedException: An exception was thrown while initializing part "Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector". ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.Caching.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
```

Reply:
You'll have to add that to the Codebases as well for VS. We can dig through it together quickly when you have time

Reply:
Sure I can add it in. I was just mentioning as @dougbu had expressed concern. I may have misunderstood.

Reply:
I was concerned something strange was going on and the transitive references that _should_ have worked weren't. @NTaylorMullen's comment explained that 😺